tools/flaky-test-reporter/github_issue_test.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/google/go-github/github"
+	"github.com/knative/test-infra/tools/flaky-test-reporter/ghutil/fakeghutil"
+)
+
+var tsMapForTest = map[string]TestStat{
+	"passed": TestStat{
+		TestName: "a",
+		Passed:   []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		Failed:   []int{},
+		Skipped:  []int{},
+	},
+	"flaky": TestStat{
+		TestName: "a",
+		Passed:   []int{1, 2, 3, 4, 5, 6, 7, 8, 9},
+		Failed:   []int{0},
+		Skipped:  []int{},
+	},
+	"failed": TestStat{
+		TestName: "a",
+		Passed:   []int{},
+		Failed:   []int{0, 1, 2, 3, 4, 5, 6, 7, 8, 9},
+		Skipped:  []int{},
+	},
+	"notenoughdata": TestStat{
+		TestName: "a",
+		Passed:   []int{0, 1, 2, 3, 4, 5, 6},
+		Failed:   []int{},
+		Skipped:  []int{7, 8, 9},
+	},
+}
+
+var (
+	fakeOrg    = "fakeorg"
+	fakeRepo   = "fakerepo"
+	fakeUserID = int64(99)
+	fakeUser   = &github.User{
+		ID: &fakeUserID,
+	}
+
+	dryrun = false
+)
+
+func getFakeGithubIssueClient() *GithubIssue {
+	fg := fakeghutil.NewFakeGithubClient()
+	fg.Repos = []string{fakeRepo}
+	fg.User = fakeUser
+	return &GithubIssue{
```

Comment:
why does both githubissue and client have user handle?

Reply:
GithubIssue was designed to carry User field, and `FakeGithubClient.CreateComment` function needs to add User field to comment, so the client needs to have it.